### PR TITLE
Hardware integration

### PR DESCRIPTION
**Pull request by Team138Entropy**

### CODE REVIEW

**Review by maxh65000 — APPROVED** [review]

File: src/main/java/frc/robot/Config.java [hunk 1]
```diff
@@ -11,12 +11,14 @@
 
   public enum Key {
     // Motors
-    INTAKE__ROLLER_PORT(1),
-    STORAGE__ROLLER_PORT(2),
-    SHOOTER__ROLLER_PORT(3),
+    //DRIVE__LEFT_FRONT_PORT()
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Config.java]: Drive motors are using constants. Let's migrate them to config.

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -227,25 +233,56 @@ public void testInit() {
...
+    } else {
+      mStorage.stopBottom();
+    }
+
+    // Top storage rollers ON while button held
+    if (mOperatorInterface.isStorageRollerTopTest()) {
+      mStorage.setTopOutput(mOperatorInterface.getOperatorThrottle());
     } else {
-      mStorage.stop();
+      mStorage.stopTop();
```
Comment (jhinspeter) [inline src/main/java/frc/robot/Robot.java]: Let's change this over to `setTopOutput(0)`, and remove the `stopTop()` and `stopBottom()`

File: src/main/java/frc/robot/subsystems/Drive.java [hunk 3]
```diff
@@ -239,6 +242,22 @@ public synchronized Rotation2d getRotation() {
     return null;
   }
 
+  public void setOutputLeftBack (double output) {
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Drive.java:246]: Add function comments that clearly state these functions are for Test Mode only.

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 4]
```diff
@@ -38,6 +38,10 @@ public void stop() {
     mRoller.set(ControlMode.PercentOutput, 0);
   }
 
+  public void setOutput(double output) {
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Intake.java:42]: Add a function comment that states this function is for Test Mode only.

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 5]
```diff
@@ -105,6 +110,10 @@ public boolean isAtVelocity() {
     return mRollerVelocity > TARGET_ROLLER_VELOCITY;
   }
 
+  public void setOutput(double output) {
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Shooter.java:114]: Add a function comment that states the function is for Test Mode only.

File: src/main/java/frc/robot/subsystems/Storage.java [hunk 6]
```diff
@@ -9,17 +9,21 @@
 /** Add your docs here. */
 public class Storage extends Subsystem {
 
-  private static final int ROLLER_PORT = Config.getInstance().getInt(Key.STORAGE__ROLLER_PORT);
+  // TODO: Need to implement top roller
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Storage.java:12]: No longer a TODO

File: src/main/java/frc/robot/subsystems/Storage.java [hunk 7]
```diff
@@ -71,22 +80,41 @@ public boolean isFull() {
...
 
+  public void setBottomOutput(double output) {
+    mBottomRoller.set(ControlMode.PercentOutput, output);
+  }
+
+  public void setTopOutput(double output) {
+    mTopRoller.set(ControlMode.PercentOutput, output);
+  }
+
+  public void stopBottom() {
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Storage.java]: Not needed. Use `setBottomOutput(0)` for test functionality instead.

File: src/main/java/frc/robot/subsystems/Storage.java [hunk 8]
```diff
@@ -71,22 +80,41 @@ public boolean isFull() {
...
+    mBottomRoller.set(ControlMode.PercentOutput, 0);
+  }
+
+  public void stopTop() {
```
Comment (jhinspeter) [inline src/main/java/frc/robot/subsystems/Storage.java]: Not needed. Use `setTopOutput(0)` for test functionality instead.

**Review by jhinspeter — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (Team138Entropy) [reply src/main/java/frc/robot/Robot.java]: Done

File: src/main/java/frc/robot/subsystems/Drive.java (hunk 3 above)
Comment (Team138Entropy) [reply src/main/java/frc/robot/subsystems/Drive.java:246]: All set

File: src/main/java/frc/robot/subsystems/Intake.java (hunk 4 above)
Comment (Team138Entropy) [reply src/main/java/frc/robot/subsystems/Intake.java:42]: Sure thing

File: src/main/java/frc/robot/subsystems/Shooter.java (hunk 5 above)
Comment (Team138Entropy) [reply src/main/java/frc/robot/subsystems/Shooter.java:114]: Done deal

File: src/main/java/frc/robot/subsystems/Storage.java (hunk 6 above)
Comment (Team138Entropy) [reply src/main/java/frc/robot/subsystems/Storage.java:12]: TODO? More like GONE

File: src/main/java/frc/robot/subsystems/Storage.java (hunk 7 above)
Comment (Team138Entropy) [reply src/main/java/frc/robot/subsystems/Storage.java]: Yeeted

File: src/main/java/frc/robot/subsystems/Storage.java (hunk 8 above)
Comment (Team138Entropy) [reply src/main/java/frc/robot/subsystems/Storage.java]: Excommunicated faster than Martin Luther

File: src/main/java/frc/robot/Config.java (hunk 1 above)
Comment (Team138Entropy) [reply src/main/java/frc/robot/Config.java]: Migrated like the Canadian Goose

**Review by jhinspeter — APPROVED** [review]

**Pull request merged**